bin/q.js

```diff
@@ -1,12 +1,84 @@
 #!/usr/bin/env node
 
-const program = require("commander");
+const { program } = require("commander");
+const chalk = require("chalk");
+const errorColor = chalk.red;
 const version = require("../package.json").version;
+const runServer = require("./commands/server.js");
+const newToolOrServer = require("./commands/newToolOrServer.js");
+const updateItem = require("./commands/updateItem/updateItem.js");
 
-program
-  .version(version)
-  .description("Q Toolbox cli")
-  .command("new-tool", "bootstrap a new tool")
-  .command("new-server", "bootstrap a new Q server implementation")
-  .command("server", "start a simple server mocking a Q server for development")
-  .parse(process.argv);
+async function main() {
+  program.version(version).description("Q Toolbox cli");
+
+  program
+    .command("server")
+    .option("-p, --port [port]", "the port to start the server on")
+    .option(
+      "-t, --target [target]",
+      "the target being used for rendering info route"
+    )
+    .option(
+      "-b, --tool-base-url [url]",
+      "the tool base url being set in rendering info route"
+    )
+    .option(
+      "-c, --config [config]",
+      "the config file name in which additional rendering info and tool runtime config is specified"
+    )
+    .description("start a simple server mocking a Q server for development")
+    .action(async () => {
+      await runServer(program);
+    });
+
+  program
+    .command("new-server")
+    .option(
+      "-d, --dir [path]",
+      "the base directory to bootstrap the new Q server implementation in, defaults to the name of Q server implementation"
+    )
+    .description("bootstrap a new Q server implementation")
+    .action(async () => {
+      const name = program.args[1];
+      if (!name) {
+        console.error(errorColor("no servername given"));
+        process.exit(1);
+      }
+      const baseDir = program.dir || name;
+      await newToolOrServer("server", name, baseDir);
+    });
+
+  program
+    .command("new-tool")
+    .option(
+      "-d, --dir <path>",
+      "the base directory to bootstrap the new tool in, defaults to the tools name"
+    )
+    .description("bootstrap a new tool")
+    .action(async () => {
```

Comment:
General remark also valid for `new-server` and `server` (`update-item` already uses the right parameter):
In order to use action based subcommands (instead of stand alone subcommands like before) you can (and should use) the object of the command instead of the `program` object to get the right results for args and specified options. Like this:
```
.action(async (commandObject) => {
      const name = commandObject.args[0];
      if (!name) {
        console.error(errorColor("no toolname given"));
        process.exit(1);
      }
      const baseDir = commandObject.dir || name;
      await newToolOrServer("tool", name, baseDir);
    });
``` 
One example: If you execute `Q new-tool -d /some/path/` while using `program.args` you will get a bootstrapped tool with the name "-d", because `program.args` includes all arguments i.e. `[ 'new-tool', '-d', '/some/path/' ]`. If you are using the command object itself, you'll get only what is relevant for that command i.e. `commandObject.args` = `[]` and therefore an error because no name is specified.
Same applies for all specified options like the directory in that command. That's part of the command object but not of the program object
https://www.npmjs.com/package/commander#action-handler-subcommands

Reply:
Ah I see. This is fixed in https://github.com/nzzdev/Q-cli/pull/30/commits/901e5d9c606a50feaee1bd1df5a7d2cb0affd04a